semantics/language/execution/stmt/return.k

```diff
@@ -51,8 +52,8 @@ module C-STMT-RETURN
                     | "Return''" "(" K ")"
      context Return'(HOLE => reval(HOLE)) [result(RValue)]
 
-     /* recursively exist each Block wrapped in BlockStatement */
-     rule <k> (.K => exitRestrictScope(BlockNum) ~> deleteLocals ~> popBlock)
+     /* recursively exits each Block wrapped in BlockStatement */
+     rule <k> (.K => exitRestrictScope(BlockNum) ~> popBlock)
```

Comment:
I seem to recall Andrei added this because he needed it. Why was it removed? also, small items like this in a large diff are quite critical but very hard to notice, and I can't necessarily guarantee that I will notice all of them. Can you remember any other cases in this diff where you made changes that were not immediately related to the task you were working on? it could be quite important.